Trip.py

```diff
@@ -0,0 +1,54 @@
+from collections import defaultdict
+from Leg import Leg
+from string import lower
+from QueryParser import QueryParser
+
+
+class Trip:
+    """docstring for Trip"""
+
+    def __init__(self, query_parser, acceptable_modes):
+
+        self._query_parser = query_parser
+
+        # {'walk': True, 'drive': True, 'bike': True, 'skate': False, 'ride': True}
+        self._acceptable_modes = acceptable_modes
+
+        self._legs = defaultdict(Leg)
+        self._build_legs()
+
+    def _build_legs(self):
+        # the end coordinates of the last stop are the start coordinates of the next segment
+        for i, segment in enumerate(self._query_parser.parse_segments()):
+            if segment['travel_mode'] == 'WALKING':
+                qp = QueryParser(segment['start_location'],
+                                 segment['end_location'],
+                                 'walking',
```

Comment:
right, i see the `self._acceptable_modes` part (line 29), but I'm talking about how the `qp = QueryParser(...)` functionality is duplicated.
```
qp = QueryParser(..., 'walking', ...)
```
is the same as
```
qp = QueryParser(..., lower(segment['travel_mode']), ...)
```

I think only the assignment `self._legs[i]` should be inside of the if/else logic.